search-engines.lisp

```diff
@@ -622,6 +622,243 @@ REQUEST-ARGS is a list of args to pass to request function."
                             (:month "month")
                             (:year "year"))))
 
+(serapeum:export-always 'startpage-image-size)
+(declaim (ftype (function (string) string) startpage-image-size))
+(defun startpage-image-size (input)
+  (let ((parsed-int (parse-integer input :junk-allowed t)))
+    (if (plusp parsed-int)
+        (format "~s" nil parsed-int)
+        (progn (log:warn "The value specified for IMAGES-SIZE-EXACT-WIDTH or IMAGES-SIZE-EXACT-HEIGHT
+is not a positive integer. Defaulting to empty value")
+         ""))))
+
+(declaim (ftype (function (string) string) startpage-settings-string))
+(defun startpage-settings-string (input)
+  (let* ((parsed-dec-int (parse-integer input :junk-allowed t :radix 16))
+         (parsed-hex-int-string (format nil "~X" parsed-dec-int)))
```

Comment:
Why re-parse if you can just check the length of the string? I.e.
``` lisp
(defun startpage-settings-string (input)
     (if (eql (length input) 160) ; The settings string should be 160 characters long.
         input
         (progn (log:warn "The value specified for SETTINGS-STRING is not valid.
  Defaulting to empty value")
          ""))))
```

Reply:
I've replaced this with a regexp.